3.Tests/UI/Hooks/ProjectHooks.cs

```diff
@@ -23,6 +24,12 @@ public ProjectHooks(ScenarioContext scenarioContext)
         _projectName = IdHelper.GetNewId();
     }
 
+    [AfterScenario]
+    public void SessionDisposal()
+    {
+        GenericWebDriver.Dispose();
+    }
+
```

Comment:
Isn't it better to define this Disposal method in a hook class and just call it in the gherkin file?
Instead of overwriting it everywhere?